File: server/bussines/hotel.js
const hotelDao = require("../dao/hotelDao.js");

const crearHotel = function(req, res){
    let hotel = 
    {
        direccion : req.body.direccion,
        region : req.body.region,
        telefono : req.body.telefono,
        administrador : parseInt(req.body.administrador)
    };
    hotelDao.crearHotel(hotel).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarHotel = function(req, res){
    let hotel = 
    {
        id : parseInt(req.body.id),
        direccion : req.body.direccion,
        region : req.body.region,
        telefono : req.body.telefono,
        administrador : parseInt(req.body.administrador)
    };
    hotelDao.actualizarHotel(hotel).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const actualizarEstadoHotel = function(req, res){
    let hotel = 
    {
        id : parseInt(req.body.id),
        estado : req.body.estado
    };
    hotelDao.actualizarEstadoHotel(hotel).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerHotelPorId = function(req, res){
    let hotel = 
    {
        id : parseInt(req.body.id)
    };
    hotelDao.obtenerHotelPorId(hotel).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerHoteles = function(req, res){
    hotelDao.obtenerHoteles().then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

module.exports.crearHotel = crearHotel;
module.exports.actualizarHotel = actualizarHotel;
module.exports.actualizarEstadoHotel = actualizarEstadoHotel;
module.exports.obtenerHoteles = obtenerHoteles;
module.exports.obtenerHotelPorId = obtenerHotelPorId;

File: Desktop/Desktop.Data/Hotel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class Hotel
    {
        private string _url;
        public int ID { get; set; }
        public string Direccion { get; set; }
        public string Region { get; set; }
        public long Telefono { get; set; }
        public bool Estado { get; set; }
        public string Administrador { get; set; }

        public Hotel()
        {
            _url = "http://127.0.0.1:3000/";
        }
        public bool CrearHotel(Hotel hotel)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "direccion", hotel.Direccion },
                { "region", hotel.Region },
                { "telefono", hotel.Telefono.ToString() },
                { "administrador", hotel.Administrador },
            };
            return Conexion.Post(queryParams, "crearHotel"); ;
        }

        public bool ActualizarHotel(Hotel hotel)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", hotel.ID.ToString() },
                { "direccion", hotel.Direccion },
                { "region", hotel.Region },
                { "telefono", hotel.Telefono.ToString() },
                { "administrador", hotel.Administrador },
            };
            return Conexion.Post(queryParams, "actualizarHotel"); ;
        }

        public bool ActualizarEstadoHotel(int id, bool estado)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", id.ToString() },
                { "estado", estado ? "1" : "0" },
            };
            return Conexion.Post(queryParams, "actualizarEstadoHotel"); ;
        }

        public Hotel ObtenerHotelPorId(int id)
        {
            Conexion.Url = _url;
            var url = new UriBuilder(_url);
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string> { { "id", id.ToString() } };
            var content = new FormUrlEncodedContent(queryParams);
            var res = Conexion.Cliente.PostAsync(_url + "obtenerHotelPorId", content).Result
                .Content.ReadAsStringAsync().Result;
            Hotel hotel = JsonConvert.DeserializeObject<List<Hotel>>(res)[0];
            return hotel;
        }

        public List<Hotel> ObtenerHoteles()
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var res = Conexion.Cliente.GetStringAsync(_url + "obtenerHoteles").Result;
            List<Hotel> empresas = JsonConvert.DeserializeObject<List<Hotel>>(res);
            return empresas;
        }
    }
}
